code/pilotfile/JSONFileHandler.cpp

```diff
@@ -4,6 +4,11 @@
 #include "libs/jansson.h"
 #include "parse/parselo.h"
 
+// it's kinda tricky to #include "utf.h" here, so we forward-declare this, but note that it's a C function
+extern "C" {
+	extern int utf8_check_string(const char *string, size_t length);
```

Comment:
What are the issues with including that header? We should avoid doing library declarations on our own if at all possible. The contents of this function also seem to be more or less the same as `utf8::find_invalid` so if including the header doesn't work maybe you could use that function instead?

Reply:
Just that it's a very convoluted path.  But I'll try that alternative.